extensions/datastores/kudu/src/main/java/org/locationtech/geowave/datastore/kudu/operations/KuduOperations.java

```diff
@@ -280,31 +285,41 @@ private boolean createIndexTable(final String indexName, int numPartitions) {
   public <T> KuduRangeRead<T> getKuduRangeRead(
       final String indexName,
       final short[] adapterIds,
-      final byte[][] dataIds,
       final Collection<SinglePartitionQueryRanges> ranges,
       final boolean rowMerging,
       final GeoWaveRowIteratorTransformer<T> rowTransformer,
       final Predicate<GeoWaveRow> rowFilter,
       final boolean visibilityEnabled) throws KuduException {
-    boolean isDataIndex = DataIndexUtils.isDataIndex(indexName);
     KuduTable table = getTable(indexName);
     return new KuduRangeRead<T>(
         ranges,
         adapterIds,
-        dataIds,
         table,
         this,
-        isDataIndex,
         visibilityEnabled,
         rowFilter,
         rowTransformer,
         rowMerging);
   }
 
+  public <T> KuduDataIndexRead<T> getKuduDataIndexRead(
+      final String indexName,
+      final short adapterId,
+      final byte[][] dataIds,
+      final Predicate<GeoWaveRow> rowFilter,
+      final boolean visibilityEnabled) throws KuduException {
+    KuduTable table = getTable(indexName);
+    return new KuduDataIndexRead<T>(adapterId, dataIds, table, this, visibilityEnabled, rowFilter);
```

Comment:
We can omit the type parameter `T` here

Reply:
It says 'KuduDataIndexRead is a raw type. References to generic type KuduDataIndexRead<T> should be parameterized'

Reply:
You should be able to remove the `T` and keep the `<>`. Java will automatically infer the type parameter.